app/models/repaymentHistory/RepaymentHistoryUtils.scala

```diff
@@ -133,14 +133,22 @@ object RepaymentHistoryUtils {
     )
   }
 
-  private def creditEntry(payment: Payment, isAgent: Boolean, isPoaReconciliationCredit: Boolean = false)(implicit messages: Messages, dateServiceInterface: DateServiceInterface): PaymentHistoryEntry = {
-    val creditType = payment.creditType.getOrElse(throw MissingFieldException("Credit type"))
-    PaymentHistoryEntry(
-      date = payment.dueDate.getOrElse(throw MissingFieldException(s"Payment Due Date - ${creditType.getClass.getSimpleName}")),
+  private def creditEntry(payment: Payment, isAgent: Boolean, isPoaReconciliationCredit: Boolean = false)(implicit messages: Messages, dateServiceInterface: DateServiceInterface): Either[Throwable, PaymentHistoryEntry] = {
+    for {
+      creditType <- payment.creditType.toRight(MissingFieldException("Credit type"))
```

Comment:
* We are suppressing anything in the Left side in the calling method => 
**filterPayment =**> 
creditEntry(payment, isAgent).toOption
* What we should do instead, is to convert **filterPayment** to return **Either[Throwable, PaymentHistoryEntry]** as well
and handle result in combinePaymentHistoryData (log error and throw error)